dht.go

```diff
@@ -678,7 +681,10 @@ func (dht *IpfsDHT) fixRTIfNeeded() {
 }
 
 // FindLocal looks for a peer with a given ID connected to this dht and returns the peer and the table it was found in.
-func (dht *IpfsDHT) FindLocal(id peer.ID) peer.AddrInfo {
+func (dht *IpfsDHT) FindLocal(ctx context.Context, id peer.ID) peer.AddrInfo {
+	_, span := internal.StartSpan(ctx, "IpfsDHT.FindLocal", trace.WithAttributes(attribute.Stringer("PeerID", id)))
+	defer span.End()
+
```

Comment:
This method is doing IO w/ the peerstore, ex suppose you are using a pstoreds (datastore-backed peerstore), and you are using some remote datastore like S3 or postgres, it would be nice to know how much the latency of this is contributing to the latency of the caller.

(It is a design flaw of the peerstore that it does not take a context, but at the very least this provides a measurement of how long FindLocal takes).

Reply:
It's nice to have entry points into an API covered because this helps a lot with debugging the big picture when looking at the traces.

If this is hot code we could remove it.